packages/data/lib/repository/auth_repository.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:domain/model/firebase_user_email.dart';
+import 'package:domain/repository/auth_repository.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter_facebook_auth/flutter_facebook_auth.dart';
+import 'package:google_sign_in/google_sign_in.dart';
+
+const _usersCollection = 'star_movie_users';
+
+class AuthRepositoryImpl implements AuthRepository {
+  final FirebaseAuth firebaseAuth;
+  final FirebaseFirestore firebaseFirestore;
+
+  const AuthRepositoryImpl(
+    this.firebaseAuth,
+    this.firebaseFirestore,
+  );
+
+  @override
+  Future<List<UserEmailPass>> fetchUsers() async {
+    final users = await firebaseFirestore.collection(_usersCollection).get();
+    return users.docs.map((e) => UserEmailPass.fromJson(e.data())).toList();
+  }
```

Comment:
we definitely don't need this method. what if we have 100 million users? instead of fetching whole table we should execute the query, like instead of

select * from users;

we should use

select * from users where email = 'email' AND password = 'password'